packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -2332,7 +2330,6 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
 
   void _onCursorColorTick() {
     renderEditable.cursorColor = widget.cursorColor.withOpacity(_cursorBlinkOpacityController.value);
-    _cursorVisibilityNotifier.value = widget.showCursor && _cursorBlinkOpacityController.value > 0;
```

Comment:
@justinmc @LongCatIsLooong 
not entirely sure whether this will be hurt the performance or improve the performance thing setting it false will cause a rebuild on the caret

Reply:
`_cursorVisibilityNotifier.value` only tells the custom painter to repaint so I don't think this does unnecessary rebuilds.